src/search.cpp

```diff
@@ -1425,9 +1425,9 @@ Value Search::Worker::search(
                        depth, bestMove, unadjustedStaticEval, tt.generation());
 
     // Adjust correction history
-    if (!ss->inCheck && (!bestMove || !pos.capture(bestMove))
-        && !(bestValue >= beta && bestValue <= ss->staticEval)
-        && !(!bestMove && bestValue >= ss->staticEval))
+    if (!ss->inCheck && !(bestMove && pos.capture(bestMove))
+        && ((bestValue < ss->staticEval && bestValue < beta )   // negative correction & no fail high
+         || (bestValue > ss->staticEval && bestMove)))          // positive correction & no fail low
```

Comment:
```c++
// Do not apply negative correction when failing high
// Do not apply positive correction when failing low
```
Do you think this might be a clearer rewording of the comments above?

Reply:
i prefer the initial version, short and concise